llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -1169,6 +1169,8 @@ instCombineSVEVectorFuseMulAddSub(InstCombiner &IC, IntrinsicInst &II,
     AddendOp = II.getOperand(2);
     Mul = II.getOperand(1);
   }
+  if (match(P, m_ZeroInt()))
```

Comment:
Are we sure that this is going have the llvm intrinsic as [llvm-ir-instrinsinc]_u?
Because only for **_u** llvm-ir intrinsics that the compiler can make sure this can return undefined values.
I agree with Paul and maybe here is not the best place to do this check.